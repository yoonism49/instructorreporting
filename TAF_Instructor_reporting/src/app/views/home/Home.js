// flow weak

import React, {
  PureComponent
}                         from 'react';
import PropTypes          from 'prop-types';
import {
  AnimatedView,
  StatsCard,
  EarningGraph,
  Notifications,
  WorkProgress,
  ClassTestLog,
  TwitterFeed,
  TodoListDemo,
  TeamMatesDemo
}                         from '../../components';
import ReactDOM from 'react-dom';
import ChartistGraph from 'react-chartist';
import '../../style/chartist.css';
import HeatMap from 'react-heatmap-grid';
import $ from 'jquery';
import Collapsible from 'react-collapsible';
import { Tab, Tabs, TabList, TabPanel } from 'react-tabs';

export function getRange(count) {
  return Array.from({ length: count }, (_, i) => i);
}
export function randomData(N, max, long){
  return (
  Array.apply(null, Array(N || 5))
    .map(function(v){
      return {
        v: Math.floor(Math.random() * (max || 20)) + 1,
        label: long ? faker.name.findName() : faker.name.firstName()
      }
    })
  )
}
var formatter = function (value){
  return frmttr()(value).regular
}
export function getRandomInt(min, max) {
  return Math.floor(Math.random() * (max - min + 1)) + min;
}
export function shiftDate(date, numDays) {
  const newDate = new Date(date);
  newDate.setDate(newDate.getDate() + numDays);
  return newDate;
}
function generateRandomValues(count, date = new Date()) {
    return getRange(count).map((index) => {
      return {
        date: shiftDate(date, -index),
        count: getRandomInt(1, 150),
      };
    })
  }
class Home extends React.Component {

  static propTypes = {
    earningGraphLabels:   PropTypes.array,
    earningGraphDatasets: PropTypes.array,
    teamMatesIsFetching:  PropTypes.bool,
    teamMates:            PropTypes.arrayOf(
      PropTypes.shape({
        picture:      PropTypes.string,
        firstname:    PropTypes.string,
        lastname:     PropTypes.string,
        profile:      PropTypes.string,
        profileColor: PropTypes.oneOf(['danger', 'warning', 'info', 'success'])
      })
    ),
    actions: PropTypes.shape({
      enterHome: PropTypes.func,
      leaveHome: PropTypes.func,
      fetchEarningGraphDataIfNeeded:  PropTypes.func,
      fetchTeamMatesDataIfNeeded:     PropTypes.func
    })
  }; 
  constructor(props, context) {
    super(props, context);

    this.handleSelect = this.handleSelect.bind(this);
    this.customTitleForValue = this.customTitleForValue.bind(this);

    this.state = {
      key: 1,
      heatData:null,
      data: {
      labels: ['Topic 9', 'Topic 7', 'Topic 4', 'Topic 1'],
      series: [
        [-12, -10, -7, -5],
        [3, 5, 8, 10]
      ]
      },
      options : {
        seriesBarDistance: 0,
        reverseData: true,
        horizontalBars: true,
        axisY: {
          offset: 70
        },
        height: '350px'
      }
    };
  }
  componentWillMount() {
    const { actions: { enterHome } } = this.props;
    enterHome();
    
  }

  componentDidMount() {
    const {
      actions: {
        fetchEarningGraphDataIfNeeded,
        fetchTeamMatesDataIfNeeded
      }
    } = this.props;
    fetchEarningGraphDataIfNeeded();
    fetchTeamMatesDataIfNeeded();
  }

  componentWillUnmount() {
    const { actions: { leaveHome } } = this.props;
    leaveHome();
  }
  componentDidUpdate() {
  }
  handleSelect(key) {
    const data = this.state.data;
    const options = this.state.options;
    this.setState({ key, data, options });
    document.querySelector('.ct-chart').__chartist__.update();
  }
  reloadHeat() {
      this.setState({heatData:null});
  }
  resize = () => this.forceUpdate();
  customTitleForValue(value) {
    return value ? `You're hovering over ${value.date.toDateString()} with value ${value.count}` : null;
  }
  render() {
    const {
      teamMates,
      teamMatesIsFetching,
      earningGraphLabels,
      earningGraphDatasets
    } = this.props;
    
    const randomValues = generateRandomValues(2000);
    const xLabels = new Array(44).fill(0).map((_, i) => `Topic ${i}`);
    const yLabels = ['Class 9', 'Class 10', 'Class 13', 'Class 5', 'Class 6', 'Class 7', 'Class 3', 'Class 2', 'Class 8', 'Class 11'];
    const data = new Array(yLabels.length)
      .fill(0)
      .map(() => new Array(xLabels.length).fill(0).map(() => Math.floor(Math.random() * 100)));
    
    return(
      <AnimatedView>
       <Tabs>
    <TabList>
      <Tab>Most Recent</Tab>
      <Tab>Test History</Tab>
    </TabList>

        <TabPanel title="Most Recent">
          <div
            className="row"
            style={{marginBottom: '5px'}}>
            <h2 className="testhistory-title">Test Results:</h2>
            <div className="col-md-2 topcard-left">
             <div className="sm-st-info"><div>Class Name</div><span className="right-align-1">Class 3</span></div>
            </div>
            <div className="col-md-3 topcard">
              <div className="sm-st-info"><div>Test Name</div><span className="right-align-2">FC - Module 06</span></div>
            </div>
            <div className="col-md-1 topcard">
               <div className="sm-st-info"><div>Date Completed</div><span className="right-align-3">2018/04/11</span></div>
            </div>
            <div className="col-md-1 topcard">
               <div className="sm-st-info"><div># Finished</div><span className="right-align-4">10</span></div>
            </div>
               <div className="col-md-1 topcard">
               <div className="sm-st-info"><div># Incomplete</div><span className="right-align-5">4</span></div>
            </div>
               <div className="col-md-1 topcard">
               <div className="sm-st-info"><div># Not Start</div><span className="right-align-6">1</span></div>
            </div>
               <div className="col-md-1 topcard">
               <div className="sm-st-info"><div>Average %</div><span className="right-align-7">50%</span></div>
            </div>
               <div className="col-md-1 topcard-right">
               <div className="sm-st-info"><div>Pass %</div><span className="right-align-8">53%</span></div>
            </div>
          </div>

          <div className="row">
            <div className="col-md-10 horizontalbar-div">
             <ChartistGraph className={'ct-octave'} data={this.state.data} options={this.state.options} type={'Bar'} redraw={'true'} responsive={'true'}/>
              
        
            </div>
            
          </div>

          <div className="row">
            <div className="col-md-12">
               <Collapsible trigger="Trainee">
               <div className="Trainee">
        <p>This is the collapsible content. It can be any element or React component you like.</p>
        <p>It can even be another Collapsible component. Check out the next section!</p>
        </div>
      </Collapsible>
       <Collapsible trigger="Requirements Not Met">
      </Collapsible>      
      <Collapsible trigger="Class Test History">
             <WorkProgress />
      </Collapsible>
            </div>
     
            {/*  
              <div className="col-md-4">
              <TwitterFeed />
            </div>
            */}
          </div>
 {/*  
          <div className="row">
            <div className="col-md-5">
              <TeamMatesDemo
                isFetching={teamMatesIsFetching}
                members={teamMates}
              />
            </div>
            <div className="col-md-7">
              <TodoListDemo />
            </div>
          </div>*/}
        </TabPanel>
        <TabPanel title="Test History">
        <div className="horiz-scroll-module">
 <div
            className=" col-md-12 row horiz-model-inside"
            style={{marginBottom: '5px'}}>
            <h2 className="testhistory-title">Test History:</h2>
           <div className="topmodule-left">
               <div className="sm-st-info"onClick = {this.reloadHeat.bind(this)} ><span>Module 01</span></div>
            </div>
            <div className="topmodule">
               <div className="sm-st-info"onClick = {this.reloadHeat.bind(this)} ><span>Module 02</span></div>
            </div>
             <div className="topmodule">
               <div className="sm-st-info" onClick = {this.reloadHeat.bind(this)}><span>Module 03</span></div>
            </div>
            <div className="topmodule">
               <div className="sm-st-info" onClick = {this.reloadHeat.bind(this)}><span>Module 04</span></div>
            </div>
             <div className="topmodule">
               <div className="sm-st-info" onClick = {this.reloadHeat.bind(this)}><span>Module 05</span></div>
            </div>
            <div className="topmodule">
               <div className="sm-st-info" onClick = {this.reloadHeat.bind(this)}><span>Module 06</span></div>
            </div>
             <div className="topmodule">
               <div className="sm-st-info" onClick = {this.reloadHeat.bind(this)}><span>Module 07</span></div>
            </div>
            <div className="topmodule">
               <div className="sm-st-info" onClick = {this.reloadHeat.bind(this)}><span>Module 08</span></div>
            </div>
             <div className="topmodule">
               <div className="sm-st-info" onClick = {this.reloadHeat.bind(this)}><span>Module 09</span></div>
            </div>
            <div className="topmodule">
               <div className="sm-st-info" onClick = {this.reloadHeat.bind(this)}><span>Module 10</span></div>
            </div>
             <div className="topmodule">
               <div className="sm-st-info" onClick = {this.reloadHeat.bind(this)} ><span>Module 11</span></div>
            </div>
           </div>
          </div>

          <div className="row">
            <div className="col-md-12">
            <div className="horiz-scroll">
             <div className="heatmap-div">
        <HeatMap
              xLabels={xLabels}
              yLabels={yLabels}
              data={data}
              background={'green'}
            />
            </div>
            </div>
            </div>
          </div>

          <div className="row">
            <div className="col-md-12">
             <div className="heatmap-div-table">
              <WorkProgress />
            </div>
            </div>

          </div>

        </TabPanel>
        </Tabs>
      </AnimatedView>
    );
  }
}

export default Home;
