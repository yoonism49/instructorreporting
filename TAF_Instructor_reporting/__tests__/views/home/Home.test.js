// @flow weak

import React            from 'react';
import Home             from '../../../src/app/views/home/Home';
import renderer         from 'react-test-renderer'; // needed both for snpashot testing but also to prevent errors from enzyme
import {
  // shallow,
  mount
}                       from 'enzyme';
import {
  MemoryRouter
}                       from 'react-router';
// import teamMatesMock    from '../../../src/app/models/teamMatesMock';

// react-highlight uses findDOMNode:
// -> jest will throw errors (no way to counter that...)
jest.mock('react-highlight');
jest.mock('chart.js');

describe('Home view', () => {
  it('renders as expected', () => {
    const props = {
      earningGraphLabels: [],
      earningGraphDatasets: [],
      teamMatesIsFetching: false,
      teamMates: [],
      actions: {
        enterHome: () => {},
        leaveHome: () => {},
        fetchEarningGraphDataIfNeeded: () => {},
        fetchTeamMatesDataIfNeeded: () => {}
      }
    };
    const component = renderer.create(
      <div>
        <MemoryRouter>
          <Home {...props} />
        </MemoryRouter>
      </div>
    ).toJSON();
    expect(component).toMatchSnapshot();
  });


  it('triggers enterHome on mount', () => {
    const mockenterHome = jest.fn();
    const mockProps = {
      earningGraphLabels: [],
      earningGraphDatasets: [],
      teamMatesIsFetching: false,
      teamMates: [],
      actions: {
        enterHome: mockenterHome,
        leaveHome: () => {},
        fetchEarningGraphDataIfNeeded: () => {},
        fetchTeamMatesDataIfNeeded: () => {}
      }
    };
    /* eslint-disable no-unused-vars */
    const wrapper = mount(
      <MemoryRouter>
        <Home {...mockProps} />
      </MemoryRouter>
    );
    expect(mockenterHome.mock.calls.length).toBe(1);
  });

  it('triggers leaveHome on unMount', () => {
    const mockleaveHome = jest.fn();
    const mockProps = {
      earningGraphLabels: [],
      earningGraphDatasets: [],
      teamMatesIsFetching: false,
      teamMates: [],
      actions: {
        enterHome: () => {},
        leaveHome: mockleaveHome,
        fetchEarningGraphDataIfNeeded: () => {},
        fetchTeamMatesDataIfNeeded: () => {}
      }
    };
    const wrapper = mount(
      <MemoryRouter>
        <Home {...mockProps} />
      </MemoryRouter>
    );
    wrapper.unmount();
    expect(mockleaveHome.mock.calls.length).toBe(1);
  });
});
