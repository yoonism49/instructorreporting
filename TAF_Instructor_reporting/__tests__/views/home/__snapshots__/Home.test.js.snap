// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home view renders as expected 1`] = `
<div>
  <section
    className="content view-enter"
  >
    <div
      className="row"
      style={
        Object {
          "marginBottom": "5px",
        }
      }
    >
      <div
        className="col-md-3"
      >
        <div
          className="sm-st clearfix"
        >
          <span
            className="sm-st-icon st-red"
          >
            <i
              className="fa fa-check-square-o"
            />
          </span>
          <div
            className="sm-st-info"
          >
            <span>
              3200
            </span>
            <div>
              Total Tasks
            </div>
          </div>
        </div>
      </div>
      <div
        className="col-md-3"
      >
        <div
          className="sm-st clearfix"
        >
          <span
            className="sm-st-icon st-violet"
          >
            <i
              className="fa fa-envelope-o"
            />
          </span>
          <div
            className="sm-st-info"
          >
            <span>
              2200
            </span>
            <div>
              Total Messages
            </div>
          </div>
        </div>
      </div>
      <div
        className="col-md-3"
      >
        <div
          className="sm-st clearfix"
        >
          <span
            className="sm-st-icon st-blue"
          >
            <i
              className="fa fa-dollar"
            />
          </span>
          <div
            className="sm-st-info"
          >
            <span>
              100,320
            </span>
            <div>
              Total Profit
            </div>
          </div>
        </div>
      </div>
      <div
        className="col-md-3"
      >
        <div
          className="sm-st clearfix"
        >
          <span
            className="sm-st-icon st-green"
          >
            <i
              className="fa fa-paperclip"
            />
          </span>
          <div
            className="sm-st-info"
          >
            <span>
              4567
            </span>
            <div>
              Total Documents
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-md-8"
      >
        <section
          className="panel "
        >
          <header
            className="panel-heading"
          >
            Earning Graph
          </header>
          <div
            className="panel-body undefined"
            style={
              Object {
                "backgroundColor": "#FFF",
              }
            }
          >
            <canvas
              height="330"
              id="linechart"
              width="600"
            />
          </div>
        </section>
      </div>
      <div
        className="col-lg-4"
      >
        <section
          className="panel"
        >
          <header
            className="panel-heading"
          >
            Notifications
          </header>
          <div
            className="panel-body notification--body-container"
            id="noti-box"
          >
            <div
              className="alert alert-danger"
            >
              <button
                className="close close-sm"
                data-dismiss="alert"
                type="button"
              >
                <i
                  className="fa fa-times"
                />
              </button>
              <span>
                <strong>
                  Oh snap!
                </strong>
                Change a few things up and try submitting again.
              </span>
            </div>
            <div
              className="alert alert-success"
            >
              <button
                className="close close-sm"
                data-dismiss="alert"
                type="button"
              >
                <i
                  className="fa fa-times"
                />
              </button>
              <span>
                <strong>
                  Well done!
                </strong>
                You successfully read this important alert message.
              </span>
            </div>
            <div
              className="alert alert-info"
            >
              <button
                className="close close-sm"
                data-dismiss="alert"
                type="button"
              >
                <i
                  className="fa fa-times"
                />
              </button>
              <span>
                <strong>
                  Heads up!
                </strong>
                This alert needs your attention, but it's not super important.
              </span>
            </div>
            <div
              className="alert alert-warning"
            >
              <button
                className="close close-sm"
                data-dismiss="alert"
                type="button"
              >
                <i
                  className="fa fa-times"
                />
              </button>
              <span>
                <strong>
                  Warning!
                </strong>
                Best check yo self, you're not looking too good.
              </span>
            </div>
            <div
              className="alert alert-danger"
            >
              <button
                className="close close-sm"
                data-dismiss="alert"
                type="button"
              >
                <i
                  className="fa fa-times"
                />
              </button>
              <span>
                <strong>
                  Oh snap!
                </strong>
                Change a few things up and try submitting again.
              </span>
            </div>
          </div>
        </section>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-md-8"
      >
        <section
          className="panel"
        >
          <header
            className="panel-heading"
          >
            Work Progress
          </header>
          <div
            className="panel-body table-responsive"
          >
            <table
              className="table table-hover"
            >
              <thead>
                <tr>
                  <td>
                    #
                  </td>
                  <td>
                    Project
                  </td>
                  <td>
                    Manager
                  </td>
                  <td>
                    Deadline
                  </td>
                  <td>
                    Status
                  </td>
                  <td>
                    Progress
                  </td>
                </tr>
              </thead>
              <tbody>
                <tr>
                  <td>
                    1
                  </td>
                  <td>
                    Facebook
                  </td>
                  <td>
                    Mark
                  </td>
                  <td>
                    10/10/2014
                  </td>
                  <td>
                    <span
                      className="label label-danger"
                    >
                      in progress
                    </span>
                  </td>
                  <td>
                    <span
                      className="badge badge-info"
                    >
                      50%
                    </span>
                  </td>
                </tr>
                <tr>
                  <td>
                    2
                  </td>
                  <td>
                    Twitter
                  </td>
                  <td>
                    Evan
                  </td>
                  <td>
                    10/8/2014
                  </td>
                  <td>
                    <span
                      className="label label-success"
                    >
                      completed
                    </span>
                  </td>
                  <td>
                    <span
                      className="badge badge-success"
                    >
                      100%
                    </span>
                  </td>
                </tr>
                <tr>
                  <td>
                    3
                  </td>
                  <td>
                    Google
                  </td>
                  <td>
                    Larry
                  </td>
                  <td>
                    10/12/2014
                  </td>
                  <td>
                    <span
                      className="label label-warning"
                    >
                      in progress
                    </span>
                  </td>
                  <td>
                    <span
                      className="badge badge-warning"
                    >
                      75%
                    </span>
                  </td>
                </tr>
                <tr>
                  <td>
                    4
                  </td>
                  <td>
                    LinkedIn
                  </td>
                  <td>
                    Allen
                  </td>
                  <td>
                    10/01/2015
                  </td>
                  <td>
                    <span
                      className="label label-info"
                    >
                      in progress
                    </span>
                  </td>
                  <td>
                    <span
                      className="badge badge-info"
                    >
                      65%
                    </span>
                  </td>
                </tr>
                <tr>
                  <td>
                    5
                  </td>
                  <td>
                    Tumblr
                  </td>
                  <td>
                    David
                  </td>
                  <td>
                    01/11/2014
                  </td>
                  <td>
                    <span
                      className="label label-warning"
                    >
                      in progress
                    </span>
                  </td>
                  <td>
                    <span
                      className="badge badge-danger"
                    >
                      95%
                    </span>
                  </td>
                </tr>
                <tr>
                  <td>
                    6
                  </td>
                  <td>
                    Tesla
                  </td>
                  <td>
                    Musk
                  </td>
                  <td>
                    01/11/2014
                  </td>
                  <td>
                    <span
                      className="label label-info"
                    >
                      in progress
                    </span>
                  </td>
                  <td>
                    <span
                      className="badge badge-success"
                    >
                      95%
                    </span>
                  </td>
                </tr>
                <tr>
                  <td>
                    7
                  </td>
                  <td>
                    Ghost
                  </td>
                  <td>
                    XXX
                  </td>
                  <td>
                    01/11/2014
                  </td>
                  <td>
                    <span
                      className="label label-info"
                    >
                      in progress
                    </span>
                  </td>
                  <td>
                    <span
                      className="badge badge-success"
                    >
                      95%
                    </span>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </section>
      </div>
      <div
        className="col-md-4"
      >
        <section
          className="panel "
        >
          <header
            className="panel-heading"
          >
            Twitter Feed
          </header>
          <div
            className="panel-body undefined"
            style={
              Object {
                "backgroundColor": "#FFF",
              }
            }
          >
            <div
              className="twt-area"
            >
              <form
                action="#"
                method="post"
              >
                <textarea
                  className="form-control"
                  name="profile-tweet"
                  placeholder="Share something on Twitter.."
                  rows="3"
                />
                <div
                  className="clearfix"
                >
                  <button
                    className="btn btn-sm btn-primary pull-right"
                    type="submit"
                  >
                    <i
                      className="fa fa-twitter"
                    />
                    Tweet
                  </button>
                  <a
                    className="btn btn-link btn-icon fa fa-location-arrow"
                    data-original-title="Add Location"
                    data-placement="bottom"
                    data-toggle="tooltip"
                    href="#"
                    style={
                      Object {
                        "textDecoration": "none",
                      }
                    }
                    title=""
                  />
                  <a
                    className="btn btn-link btn-icon fa fa-camera"
                    data-original-title="Add Photo"
                    data-placement="bottom"
                    data-toggle="tooltip"
                    href="#"
                    style={
                      Object {
                        "textDecoration": "none",
                      }
                    }
                    title=""
                  />
                </div>
              </form>
            </div>
            <ul
              className="media-list"
            >
              <li
                className="media"
              >
                <a
                  className="pull-left"
                  href="#"
                >
                  <img
                    alt="Avatar"
                    className="img-circle"
                    height="64"
                    src="test-file-stub"
                    width="64"
                  />
                </a>
                <div
                  className="media-body"
                >
                  <span
                    className="text-muted pull-right"
                  >
                    <small>
                      <em>
                        30 min ago
                      </em>
                    </small>
                  </span>
                  <a
                    href="#"
                  >
                    <strong>
                      Emmet
                    </strong>
                  </a>
                  <p>
                    In hac
                    <a
                      href="#"
                    >
                      habitasse
                    </a>
                    platea dictumst. Proin ac nibh rutrum lectus rhoncus eleifend.
                    <a
                      className="text-danger"
                      href="#"
                    >
                      <strong>
                        #dev
                      </strong>
                    </a>
                  </p>
                </div>
              </li>
              <li
                className="media"
              >
                <a
                  className="pull-left"
                  href="#"
                >
                  <img
                    alt="Avatar"
                    className="img-circle"
                    height="64"
                    src="test-file-stub"
                    width="64"
                  />
                </a>
                <div
                  className="media-body"
                >
                  <span
                    className="text-muted pull-right"
                  >
                    <small>
                      <em>
                        35 min ago
                      </em>
                    </small>
                  </span>
                  <a
                    href="#"
                  >
                    <strong>
                      Jannie
                    </strong>
                  </a>
                  <p>
                    In hac
                    <a
                      href="#"
                    >
                      habitasse
                    </a>
                    platea dictumst. Proin ac nibh rutrum lectus rhoncus eleifend.
                    <a
                      className="text-danger"
                      href="#"
                    >
                      <strong>
                        #dev
                      </strong>
                    </a>
                  </p>
                </div>
              </li>
            </ul>
          </div>
        </section>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-md-5"
      >
        <section
          className="panel "
        >
          <header
            className="panel-heading"
          >
            Teammates
          </header>
          <div
            className="panel-body undefined"
            style={
              Object {
                "backgroundColor": "#FFF",
              }
            }
          >
            <ul
              className="list-group teammates"
            />
            <div
              className="panel-footer bg-white"
            >
              <button
                className="btn btn-primary btn-addon btn-sm"
                onClick={undefined}
              >
                <i
                  className="fa fa-plus"
                />
                Add Teammate
              </button>
            </div>
          </div>
        </section>
      </div>
      <div
        className="col-md-7"
      >
        <section
          className="panel tasks-widget"
        >
          <header
            className="panel-heading"
          >
            Todo list
          </header>
          <div
            className="panel-body undefined"
            style={
              Object {
                "backgroundColor": "#FFF",
              }
            }
          >
            <div
              className="task-content"
            >
              <ul
                className="task-list"
              >
                <li>
                  <div
                    className="task-checkbox"
                  >
                    <input
                      checked={false}
                      className="flat-grey list-child"
                      type="checkbox"
                    />
                  </div>
                  <div
                    className="task-title"
                  >
                    <span
                      className="task-title-sp"
                    >
                      Director is Modern Dashboard
                    </span>
                    <span
                      className="label label-success"
                    >
                      2 days
                    </span>
                    <div
                      className="pull-right hidden-phone"
                    >
                      <button
                        className="btn btn-default btn-xs"
                        onClick={[Function]}
                      >
                        <i
                          className="fa fa-pencil"
                        />
                      </button>
                    </div>
                  </div>
                </li>
                <li>
                  <div
                    className="task-checkbox"
                  >
                    <input
                      checked={false}
                      className="flat-grey list-child"
                      type="checkbox"
                    />
                  </div>
                  <div
                    className="task-title"
                  >
                    <span
                      className="task-title-sp"
                    >
                      Fully Responsive & Bootstrap 3.0.2 Compatible
                    </span>
                    <span
                      className="label label-danger"
                    >
                      Done
                    </span>
                    <div
                      className="pull-right hidden-phone"
                    >
                      <button
                        className="btn btn-default btn-xs"
                        onClick={[Function]}
                      >
                        <i
                          className="fa fa-pencil"
                        />
                      </button>
                    </div>
                  </div>
                </li>
                <li>
                  <div
                    className="task-checkbox"
                  >
                    <input
                      checked={false}
                      className="flat-grey list-child"
                      type="checkbox"
                    />
                  </div>
                  <div
                    className="task-title"
                  >
                    <span
                      className="task-title-sp"
                    >
                      Latest Design Concept
                    </span>
                    <span
                      className="label label-warning"
                    >
                      Company
                    </span>
                    <div
                      className="pull-right hidden-phone"
                    >
                      <button
                        className="btn btn-default btn-xs"
                        onClick={[Function]}
                      >
                        <i
                          className="fa fa-pencil"
                        />
                      </button>
                    </div>
                  </div>
                </li>
                <li>
                  <div
                    className="task-checkbox"
                  >
                    <input
                      checked={false}
                      className="flat-grey list-child"
                      type="checkbox"
                    />
                  </div>
                  <div
                    className="task-title"
                  >
                    <span
                      className="task-title-sp"
                    >
                      Write well documentation for this theme
                    </span>
                    <span
                      className="label label-primary"
                    >
                      3 Days
                    </span>
                    <div
                      className="pull-right hidden-phone"
                    >
                      <button
                        className="btn btn-default btn-xs"
                        onClick={[Function]}
                      >
                        <i
                          className="fa fa-pencil"
                        />
                      </button>
                    </div>
                  </div>
                </li>
                <li>
                  <div
                    className="task-checkbox"
                  >
                    <input
                      checked={false}
                      className="flat-grey list-child"
                      type="checkbox"
                    />
                  </div>
                  <div
                    className="task-title"
                  >
                    <span
                      className="task-title-sp"
                    >
                      Don't bother to download this Dashbord
                    </span>
                    <span
                      className="label label-inverse"
                    >
                      Now
                    </span>
                    <div
                      className="pull-right hidden-phone"
                    >
                      <button
                        className="btn btn-default btn-xs"
                        onClick={[Function]}
                      >
                        <i
                          className="fa fa-pencil"
                        />
                      </button>
                    </div>
                  </div>
                </li>
                <li>
                  <div
                    className="task-checkbox"
                  >
                    <input
                      checked={false}
                      className="flat-grey list-child"
                      type="checkbox"
                    />
                  </div>
                  <div
                    className="task-title"
                  >
                    <span
                      className="task-title-sp"
                    >
                      Give feedback for the template
                    </span>
                    <span
                      className="label label-success"
                    >
                      2 Days
                    </span>
                    <div
                      className="pull-right hidden-phone"
                    >
                      <button
                        className="btn btn-default btn-xs"
                        onClick={[Function]}
                      >
                        <i
                          className="fa fa-pencil"
                        />
                      </button>
                    </div>
                  </div>
                </li>
                <li>
                  <div
                    className="task-checkbox"
                  >
                    <input
                      checked={false}
                      className="flat-grey list-child"
                      type="checkbox"
                    />
                  </div>
                  <div
                    className="task-title"
                  >
                    <span
                      className="task-title-sp"
                    >
                      Tell your friends about this admin template
                    </span>
                    <span
                      className="label label-danger"
                    >
                      Now
                    </span>
                    <div
                      className="pull-right hidden-phone"
                    >
                      <button
                        className="btn btn-default btn-xs"
                        onClick={[Function]}
                      >
                        <i
                          className="fa fa-pencil"
                        />
                      </button>
                    </div>
                  </div>
                </li>
              </ul>
            </div>
            <div
              className=" add-task-row"
            >
              <a
                className="btn btn-success btn-sm pull-left"
                href="#"
              >
                Add New Tasks
              </a>
              <a
                className="btn btn-default btn-sm pull-right"
                href="#"
              >
                See All Tasks
              </a>
            </div>
          </div>
        </section>
      </div>
    </div>
  </section>
</div>
`;
